app/src/main/java/com/example/pygmyhippo/user/EventDB.java

```diff
@@ -95,4 +101,76 @@ public void onComplete(@NonNull Task<Void> task) {
                     }
                 });
     }
-}
+
+    /**
+     * This method will return the list of events that an entrant is in
+     * @author Kori
+     * @param accountID The ID of the entrant
+     * @param listener What gets called when the data is retrieved
+     */
+    public void getEntrantEvents(String accountID, DBOnCompleteListener<Event> listener) {
+        // Query modified from James' AllEventsDB
+        db.collection("Events")
+                .get()
+                .addOnCompleteListener(task -> {
+                    if (task.isSuccessful()) {
+                        // Get all the docs
+                        QuerySnapshot docs = task.getResult();
+                        Log.d("DB", String.format("Found %d Events", docs.size()));
+
+                        // Convert each doc to an event and store it only if it contains the entrant
+                        ArrayList<Event> eventList = new ArrayList<>();
+                        docs.forEach(doc -> {
+                            // Convert the doc to an event object
+                            Event event = doc.toObject(Event.class);
+
+                            // Go through the entrant list if it has
+                            if (event.getEntrants() != null) {
+                                // Note, setting the status won't alter the results of the query
+                                // because hasEntrant only matches by accountID
+                                if (event.hasEntrant(new Entrant(accountID, Entrant.EntrantStatus.waitlisted))) {
+                                    // If the current Account ID matches, then add this event to the list to return
+                                    eventList.add(event);
+                                }
+                            }
+                        });
+
+                        // Call the listener
+                        listener.OnCompleteDB(eventList, 3, DBOnCompleteFlags.SUCCESS.value);
+                    } else {
+                        Log.d("DB", "Could not get Events");
+                        listener.OnCompleteDB(new ArrayList<>(), 3, DBOnCompleteFlags.ERROR.value);
+                    }
+                });
+        }
+
+    /**
+     * Gets a long-live download uris for Picasso to get image from Firebase Storage.
+     * @author James
+     * @param url String rul with gs:// link to Firebase Storage image.
+     * @param listener DBOnCompleteListener to call when query completes.
+     */
+    public void getImageDownloadUrl(String url, StorageOnCompleteListener<Uri> listener) {
```

Comment:
could this also make sense in the image storage class in the database folder

Reply:
Yah I'll move all this image stuff to the image file... I forgot it existed